src/polymer/pseudo-element-scanner.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as dom5 from 'dom5';
+import {ASTNode} from 'parse5';
+
+import * as jsdoc from '../javascript/jsdoc';
+import {annotateElementHeader} from './docs';
+
+import {HtmlVisitor, ParsedHtmlDocument} from '../html/html-document';
+import {HtmlScanner} from '../html/html-scanner';
+import {ScannedPolymerElement} from './polymer-element';
+
+export class PseudoElementScanner implements HtmlScanner {
```

Comment:
A pseudo element is probably a weird enough concept to be worth adding a comment about, and here seems like the natural place. I'd go with something like:

```javascript
/**
 * A Polymer pseudo-element is an element that is declared in an unusual way, such
 * that the analyzer couldn't normally analyze it, so instead it is declared in
 * comments.
 */
```